Feature/frontend/theme/overhaul

danpolanco opened the pull request:
I still have some minor changes I want to make. I also would like your opinion on `resources/templates/frames`. I made that folder as I was going along to keep what I was working on separate and easy to debug.

Let me know what you think!

Todo:
- [x] Fix login screen text (it's invisible in firefox and partially in safari)
- [x] Remove account information from top bar
- [x] test all functions on live server to make sure I didn't break anything
- [x] clean up code (since I'm new to angular, css, html)
- [x] ~~rebase all changes into one nice commit?~~

danpolanco committed:
Update gitignore to include vim / other stuff

danpolanco committed:
Update formatting slightly for #includes

danpolanco committed:
Add new theme source code

danpolanco committed:
Add top navbar from tui2tone theme

danpolanco committed:
Move frames to their own folder and rename to html

danpolanco committed:
Create fake main view frame

danpolanco committed:
Remove old css styling

danpolanco committed:
Add icon information for views

danpolanco committed:
Move frames back to templates due to inclusion error

danpolanco committed:
Add enough code to resemble tui2tone without ext js

danpolanco committed:
Add js from tui2tone theme

danpolanco committed:
Add active class for sidebar

For some reason, the sidebar requires a class="active"
and adding that fixes it.

danpolanco committed:
Add ng-repeat in sidebar plus active tab

danpolanco committed:
Update branch to latest changes

danpolanco committed:
Finally fix broken sidebar ng-view goto

danpolanco committed:
Add login page styling; needs tweaks

danpolanco committed:
Fix login input so text is visible

danpolanco committed:
Remove account from topbar

itzg commented:
I pulled down your branch and ran it...great job! I only found two visual things I'd like you to address:

![image](https://cloud.githubusercontent.com/assets/988985/12368777/4bbf6c72-bbb4-11e5-88df-1c9dea92ecd8.png)

![image](https://cloud.githubusercontent.com/assets/988985/12368779/50a6ccda-bbb4-11e5-9b47-d2edb4b8baa1.png)

I haven't looked at the code yet, so let me know if you want me to wait until you knock out your cleanup TODOs first. Or I can look before then and give you advice, if you want.

DanTheColoradan commented:
Please take a look :) cuz I dunno where to start.

Sent from my mobile so sorry for the being curt!

On Jan 15, 2016, at 17:19, Geoff Bourne <notifications@github.com> wrote:

> I pulled down your branch and ran it...great job! I only found two visual things I'd like you to address:
> 
> 
> 
> 
> 
> I haven't looked at the code yet, so let me know if you want me to wait until you knock out your cleanup TODOs first. Or I can look before then and give you advice, if you want.
> 
> —
> Reply to this email directly or view it on GitHub.
>

itzg commented on src/main/resources/static/js/alert.js:
```diff
@@ -0,0 +1,7 @@
+$(function() {
```
These various js files came along with the theme, right? Can you put them in a sub-directory to emphasize that they are not typically for us to maintain? Ideally these would have been pulled via a webjar and we wouldn't need to source control them in the first place. Any possibility of that? 

With that said, these also don't look very "angular". It's an artificial line to draw, but I'd really like to stick to Angular-centric widgets/modules, such as those available at

http://ngmodules.org/

danpolanco replied:
I'll try and remove a whole bunch of them. Most we won't need and those that we do, I'll see if I can find them in a web jar.

itzg replied:
...I took a quick look and didn't find a webjar for it. Is there an NPM/Bower for it?

In any case, prune the cruft and place in a sub-directory will be good enough.

danpolanco replied:
Sort of done. Only navbar.js left.

itzg commented on src/main/resources/static/css/app.css:
```diff
@@ -1,143 +0,0 @@
-body {
```
This makes me nervous...I'm wondering where all my precious CSS went...but I'm going to patiently wait to see it re-appear further down.

danpolanco replied:
You won't find it O_O but that's because I removed it at one point to get the admin theme up and running. Let me know what you'd like back.

itzg replied:
Since the UI looks fine, then apparently I don't know what I'm missing :) other than the mod drop area. That took some CSS-fu to get laid out right.

itzg commented on src/main/resources/static/js/index.js:
```diff
@@ -0,0 +1,134 @@
+$(function() {
```
A third party library naming their charting "index" seems like they're asking for a filename conflict. All the more reason to put this in a sub-directory.

danpolanco replied:
Done.

itzg commented on src/main/resources/static/js/routes.js:
```diff
@@ -1,44 +1,56 @@
-            })
-            .otherwise('/view')
-    })
+.config(function ($routeProvider) {
```
Please revert these whitespace changes of space to tabs. FYI I am using IntelliJ's default JS indentation engine.

danpolanco replied:
Yeah. I wasn't sure about that. I went with 4 space for most things I wasn't sure about. I'll go ahead and change them back.

itzg replied:
Admittedly I need to capture the indentation, etc specs in an [EditorConfig](http://editorconfig.org/)...but I learned about that a couple weeks ago :)

danpolanco replied:
That's new to me! Cool.

itzg replied:
...and it definitely works. I brought a contractor's code into my IntelliJ and it honored the EditorConfig with no manual intervention.

danpolanco replied:
Done.

itzg commented on src/main/resources/templates/frames/footer.html:
```diff
@@ -0,0 +1,5 @@
+<footer class="app-footer">
```
I know this was an unplanned collision, but please incorporate my "by-line" template that I created. Of most importance:
- A link to the Github repo for support
- Deploy-time-configurable kudos to the hosting provider -- only Carina so far

itzg commented on src/main/resources/templates/ftl-bits/app-js.ftl:
```diff
@@ -1,13 +1,21 @@
+<script type="text/javascript" src="js/button.js"></script>
+<script type="text/javascript" src="js/card.js"></script>
+<script type="text/javascript" src="js/theming.js"></script>
+<script type="text/javascript" src="js/chartjs.js"></script>
```
Can you trim this down to only those theme js files that are absolutely necessary. For example, for charting, I will go with [Angular-nvd3](http://krispo.github.io/angular-nvd3/#/) .

danpolanco replied:
Yeah. I wasn't sure if there is a way to figure that out without deleting line-by-line and trying them out. Have any idea?

itzg replied:
Unfortunately, brute force remove-and-try is what I would do...or the inverse, knowing what features of the theme are used, only add those in. That's what I do for all the optional core module of Angular.

itzg commented on src/main/resources/templates/ftl-bits/by-line.ftl:
```diff
@@ -1,17 +0,0 @@
-<div class="by-line text-muted">
```
Please bring this back

danpolanco replied:
Done.

itzg commented on src/main/resources/templates/ftl-bits/ext-css.ftl:
```diff
@@ -1,7 +1,17 @@
+<link rel="stylesheet" type="text/css" href="webjars/animate.css/3.5.0/animate.css">
+<link rel="stylesheet" type="text/css" href="webjars/bootstrap-switch/3.3.2/dist/css/bootstrap3/bootstrap-switch.min.css">
+<link rel="stylesheet" type="text/css" href="webjars/checkbox3/0.1.1/dist/checkbox3.min.css">
+<link rel="stylesheet" type="text/css" href="webjars/DataTables/1.10.10/media/css/jquery.dataTables.min.css">
```
Similar comment about trimming down theme css that aren't needed. For data tables I will use [ui-grid](http://ui-grid.info/).

danpolanco replied:
Done.

itzg commented on src/main/resources/templates/ftl-bits/app-js.ftl:
```diff
@@ -1,13 +1,21 @@
+<script type="text/javascript" src="js/view-containers-ctrl.js"></script>
+<script type="text/javascript" src="js/upload-mod-ctrl.js"></script>
+<script type="text/javascript" src="js/manage-mods-ctrl.js"></script>
+<!-- Tui2tone admin v.2 bootstrap -->
```
Don't put third party js in `app-js.ftl`. Just files created by us, please. These can go in `ext-js.ftl`

itzg commented:
Sorry for all the comments. This turned out to be a good conduit to voice some contributor guidelines that were in my head.

DanTheColoradan commented:
No worries. Now time for me to fix all of them :dash:

danpolanco committed:
Move theme css/js to ext ftl

danpolanco committed:
Add space between current container labels

danpolanco committed:
Stretch and center upload mod button

danpolanco committed:
Revert tabs to spaces due to accidental change

itzg commented:
I just noticed "Status" needs space after the label:
![image](https://cloud.githubusercontent.com/assets/988985/12373337/c3e5ca66-bc3b-11e5-9c53-c802bf9b89b5.png)

Also, I tried out the interface on my iPhone and it's a lot more usable than it was before; **however**, it seems to take a few taps on "Create Container" to go there and even then the sidebar doesn't slide back automatically. I haven't investigated any yet to see if the click is too small or what. It's also a pain since Chrome on my iPhone doesn't show if there's any JS errors occurring. For what it's worth here's a screen grab:

![img_5586](https://cloud.githubusercontent.com/assets/988985/12373347/18665eac-bc3c-11e5-8265-d400c2a71526.PNG)

DanTheColoradan commented:
Thanks for testing it out!

FYI not related to the above convo, there are more style changes I want to make that I'm leaving alone until after a successful pull request. For example:

![screen shot 2016-01-16 at 9 50 06 am](https://cloud.githubusercontent.com/assets/3826143/12373440/97e68b8a-bc36-11e5-9680-7c2fa5cce755.png)

the white border will either go away, or change to look more natural. Stuff like that.

itzg commented:
Sounds like a good plan.

danpolanco committed:
Move tui2tone javascript and remove unnecessary webjars

danpolanco committed:
Remove unnecessary tui2tone javascript

danpolanco committed:
Add back template footer with 'This deployment' image

danpolanco committed:
Recreate frames folder and change ftl-bits to includes/

DanTheColoradan commented:
I haven't had a change to fix the touch / sidebar slide back yet, but the rest of the issues are pretty well done.

danpolanco committed:
Remove atm extra functions from tui2tone navbar.js

danpolanco committed:
Reindent theme.css to match the rest of the code

danpolanco committed:
Remove buggy asterisks from theme.css

danpolanco committed:
Replace col-md with col-sm

Supposedly it is easier to start with sm and then scale up if needed.

danpolanco committed:
Change images to img to lineup with most other projects

danpolanco committed:
Use code inspection to remove unused java imports

danpolanco committed:
Use code inspection to change a variable to local

danpolanco committed:
Use code inspection to remove redundant px

danpolanco committed:
Use code inspection to use shorthand in CSS

danpolanco committed:
Use code inspection to remove overwritten property values

danpolanco committed:
Use code inspection to change some syntax

danpolanco committed:
Use code inspection to fix small HTML errors

danpolanco committed:
Use code inspection to fix required=t to just required

DanTheColoradan commented:
I looked and I'm not sure about the touch / sidebar slide back on mobile issue. Hopefully it'll become apparent as I keep working with the theme.

Otherwise, I think this puppy is ready to go. Since it hasn't been merged yet, I'm going to continue to work on it. I don't really want to, but I think it's time to go through all the css and figure out what is actually needed. Right now, it's a mess :disappointed:

DanTheColoradan commented:
Cancel that. It's actually ok for now.

itzg commented:
Since that file is focused on the theme, then it's okay to leave all that in there. Being cascading we can then include a style sheet after the theme for specific tweaks. 

Now, if there's any modifications you had to make, then put back the original theme css and then move your  mods to the new cascaded style sheet.

DanTheColoradan commented:
Cool. The only changes I made were just bugs (e.g. width: 90px and width: 100% both present).

danpolanco committed:
Fix small UI bug on top bar button disappearing

danpolanco committed:
Remove unnecessary right button on topbar

DanTheColoradan commented:
Whoops! Broke the login page. Gotta figure out where I did that.

danpolanco committed:
Rename view.html to ftl like all other frames have been

DanTheColoradan commented:
Very weird. I have firefox on private browsing so it should start fresh. When I first click the login button it goes to this:

![screen shot 2016-01-16 at 12 47 46 pm](https://cloud.githubusercontent.com/assets/3826143/12374121/751751c0-bc4f-11e5-9510-f842f8117a63.png)

If I press the back button and click login again, it goes to the correct page...

itzg commented:
To baseline weird redirect scenarios like that, enter `http://localhost:8080/` in the address bar and press Enter/Return to confirm you're starting from that known point.

danpolanco committed:
Cleanup login.ftl in attempt to fix bad redirect

DanTheColoradan commented:
No dice :-/ I'm not sure what's up.

DanTheColoradan commented:
Huh. Nvm. When I remove:

```html
<!--suppress HtmlUnknownTarget -->
<script type="text/javascript" src="vendors/tui2tone/navbar.js"></script>
```
From the ext-js.ftl, it works right.

DanTheColoradan commented:
So I guess remove tui2tone/navbar.js and replace its function with something from ngmodules.org?

danpolanco committed:
Remove the last of tui2tone javascript

The sidebar toggle button doesn't work now. But I'm going to fix that with bootstrap / angularjs.

danpolanco committed:
Revert change back to uib-progressbar

danpolanco committed:
Use UI bootstrap collapse to get close to desired collapse

itzg commented:
No that's fine to stick with tui2tone for the navbar. Since there's not a lot of Angular going on there, it's not a problem.

itzg commented:
@DanTheColoradan , where do you stand on this now? I'm happy with all the changes now, but wanted to wait to merge until I got a green light from you.

DanTheColoradan commented:
You have the green light!

I'm not entirely satisfied because I want to get rid of the last bit of js from the theme. I can do that separately though.

Let me know if you need any other final changes / reversions.

On Jan 18, 2016, at 21:20, Geoff Bourne <notifications@github.com> wrote:

> @DanTheColoradan , where do you stand on this now? I'm happy with all the changes now, but wanted to wait to merge until I got a green light from you.
> 
> —
> Reply to this email directly or view it on GitHub.
>

itzg commented:
Check it out @DanTheColoradan , your changes are live!

https://mccy.itzg.me

DanTheColoradan commented:
Woot :D thx for the update!


————
Daniel L. Polanco
Fledgling Programmer | Perl | C++
719-422-3765 | Website | GitHub | Twitter | Safe Email
————

On Jan 19, 2016, at 20:36, Geoff Bourne <notifications@github.com> wrote:

> Check it out @DanTheColoradan , your changes are live!
> 
> https://mccy.itzg.me
> 
> —
> Reply to this email directly or view it on GitHub.
>

sshipway commented:
I like the look of the new frontend... the mod upload (once it works :)) is much better with a progress meter.

itzg commented:
I'm liking the new look a whole lot better. There always was a progress bar there...but since it's failing now you can see it :)
